src/output.cpp

```diff
@@ -1720,6 +1721,63 @@ std::string string_format(std::string pattern, ...)
     return result;
 }
 
+std::string replace_given_name_tag(std::string input)
+{
+    size_t offset = input.find("<given_name>");
+    while (offset != std::string::npos) {
+        input.replace(offset, 12, NameGenerator::generator().getName(nameIsGivenName));
+        // capitalize the first letter of the name
+        capitalize_letter(input, offset);
+        offset = input.find("<given_name>");
+    }
+    return input;
+}
+
+std::string replace_family_name_tag(std::string input)
+{
+    size_t offset = input.find("<family_name>");
+    while (offset != std::string::npos) {
+        input.replace(offset, 13, NameGenerator::generator().getName(nameIsFamilyName));
+        // capitalize the first letter of the name
+        capitalize_letter(input, offset);
+        offset = input.find("<family_name>");
+    }
+    return input;
+}
+
+std::string replace_full_name_tag(std::string input)
+{
+    size_t offset = input.find("<full_name>");
+    while (offset != std::string::npos) {
+        input.replace(offset, 11, NameGenerator::generator().getName(nameIsFullName));
+        // capitalize the first letter of the name
+        capitalize_letter(input, offset);
+        offset = input.find("<full_name>");
+    }
+    return input;
+}
+
+std::string replace_name_tags(std::string input)
+{
+    input = replace_full_name_tag(input);
+    input = replace_family_name_tag(input);
+    input = replace_given_name_tag(input);
+    return input;
+}
+
+std::string replace_city_tag(std::string input, std::string name)
+{
+    size_t offset = input.find("<city>");
+    while (offset != std::string::npos) {
+        input.replace(offset, 6, name);
+        // capitalize the first letter of the city
```

Comment:
Why all the capitalization? Names should already have the proper form, there should be no need to do this manually. Btw. the original code in `replace_with_npc_name` capitalized the first character of the *sentence*.

Also: holy code duplication! Have you considered to create a simple `void replace(std::string &text, const std::string &pattern, const std::string &replacement); ` function that replaces all appearances of string A with string B in a given input. And just call that function `replace(text, "<city>", city_name)`?

That function could be used in many other places.